docs/02-guides/dynamic-routing.mdx

```diff
@@ -194,6 +200,69 @@ requirements.
 
 </Steps>
 
+## Batching and Prioritizing URL Matchers
+
+When multiple URL matchers are registered without any `matcherOptions`, it will
+default the batchOrder and priority to 0. This means that the order in which the
+URL matchers are registered will determine the order in which they are executed.
+
+Let's say we have the following list of URL Matchers:
+
+```bash
+A → fast API (cached)
+B → no API (static)
+C → slow API (not cached)
+D → no API (static)
+```
+
+We would ideally want this to first try to run `B` and `D` then `A` and finally
+`C`. To achieve this we can set the `batchOrder` and `priority` fields in the
+`matcherOptions` object.
+
+```ts title="extension/index.ts"
+services.DynamicRoutes.registerUrlMatcher("A", () => new UrlMatcher(), {
```

Comment:
**nitpick:** it'd be better to explicitely have different classes for matchers

**suggestion:** replace the `new UrlMatcher()` with `UrlMatcherA` / `UrlMatcherB` / `UrlMatcherC` …

```suggestion
services.DynamicRoutes.registerUrlMatcher("A", () => new UrlMatcherA(), {
```